### DISCUSSION TITLE
Using Toolchain to build

### DISCUSSION THREAD

**Discussion by andrelouw-ir** [roleoroleo/sonoff-hack#70] [Q&A]
Could you please provide a little more info on how to use the toolchain after downloading it?

I have tried to build the sonoff-hack project by using the builder/Dockerfile to build an image and then building the project in a running container from that image.

It seems like the PATH that is set at the end of the Docker file is not correct? Looks like it should add /opt/arm-sonoff-linux-uclibcgnueabi/arm-sonoff-linux-uclibcgnueabi/bin:opt/arm-sonoff-linux-uclibcgnueabi/bin (both bin folders)?

I think the image should also have the 'make' utility installed?

I cloned the sonoff-hack GIT project and passed that folder as a volume when I run a container from the Docker image:
docker run -v $(pwd):/sonoff-hack -it <image_id> bash

After fixing the PATH I was still getting errors when running the ./scripts/compile.sh script:

gcc: error trying to exec 'cc1': execvp: No such file or directory

Any idea what else might need to be done?

**Comment by roleoroleo** [comment]
Which Dockerfile do you mean?

I didn't try to compile this tree within a container and I don't know how to help you.
But for sure you need to install dependencies:
`git unzip build-essential libtool bison bison++ bisonc++ libbison-dev autoconf autotools-dev automake libssl-dev zlibc zlib1g-dev`

I'm using the following script to set the env during build:
```
#!/bin/bash

export PATH=${PATH}:/home/user/x-tools/arm-sonoff-linux-uclibcgnueabi/bin

export TARGET=arm-sonoff-linux-uclibcgnueabi
export CROSS=arm-sonoff-linux-uclibcgnueabi
export BUILD=x86_64-pc-linux-gnu

export CROSSPREFIX=${CROSS}-

export STRIP=${CROSSPREFIX}strip
export CXX=${CROSSPREFIX}g++
export CC=${CROSSPREFIX}gcc
export LD=${CROSSPREFIX}ld
export AS=${CROSSPREFIX}as
export AR=${CROSSPREFIX}ar
```

You have to adjust the first variable with your install folder.

**Reply by andrelouw-ir** [reply]
Thank you for your quick response!

Since asking this I have actually forked the project and added the build.yml workflow. That works like a charm!

**Reply by roleoroleo** [reply]
PRs are welcome.
:)
